### PR TITLE
Make the Accessible Dialog scrollable

### PR DESCRIPTION
**Pull request by kobryan0619**
I had worked on a feature so teachers can assign multiple sections content at the same time.  However in [ZenDesk](https://codeorg.zendesk.com/agent/tickets/453722) we have had a few reports of teacher not being able to hit the "submit" button because either they were too zoomed in or they had so many sections that the list was too long that it forced the submit button off the screen.  It seems like this could be an accessibility issue for anyone too zoomed it or for any content too long for the `AccessibleDialog` feature.  

This shows the issue before:
https://github.com/code-dot-org/code-dot-org/assets/24883357/5b51b9fd-d85e-4d17-a754-1a4d711dac58

This shows the fix:

https://github.com/code-dot-org/code-dot-org/assets/24883357/743dd0ce-9579-4040-9c43-5cf0ab712d93

Finally, I am tagging student-learning since I saw they have some items that used this component (esp. in HOC) to make sure this is OK with them.

## PR Checklist:

- [ ] Tests provide adequate coverage
- [ ] Privacy and Security impacts have been assessed
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

File: apps/src/templates/multiple-sections-assigner.module.scss [hunk 1]
```diff
@@ -23,17 +23,10 @@
   }
 
   .sectionListOptionsContainer {
-    &:focus {
```
Comment (kobryan0619) [inline apps/src/templates/multiple-sections-assigner.module.scss:26]: All changes in this file are just "undoing" an initial fix that forced just the sections to be scrollable.  However this fix still led to issues for teachers who were super scrolled in thus this new fix was invented.  I experimented with having two scrollers (one for sections, one for the whole dialog) but it was kinda hard to get to the "whole dialog" scroll when on the "sections scroll".  Thus I decided to eliminate this.

**Review by etaderhold — APPROVED** [review]
This looks much better!

Should we ask a designer for an opinion on the right trade-off between amount of content in the dialog box and the size of the margin below it? The margin looks maybe a touch small to my eye, but don't change it on my account.

**Review by levadadenys — APPROVED** [review]
LGTM from design system front end implementation perspective!

But yeah, might worth someone from @code-dot-org/student-labs taking a look. (if `apps/src/dance/ai/DanceAiModal.tsx` is working correctly). 
From what I see modal is starting to shrink at ~570-px screnn height
image
And starting to look broken at ~475-px screen height
image

cc @sanchitmalhotra126 @breville might worth add overwriting max-height to `DanceAiModal`?

**Comment by breville** [comment]
Some thoughts:

- I think it might be better for this particular dialog if the buttons at the bottom are always visible, and the body content is scrollable.
- As a shared component, designed for very widespread use, I'm not sure whether it makes sense to make this always scrollable, given that other users of it might want different behavior.
- We are already having to use `! important` to [apply](https://github.com/code-dot-org/code-dot-org/blob/902c0f44fad6c1e35f2049dca8f7ad0a9f67a596/apps/src/dance/ai/dance-ai-modal.module.scss#L82-L87) desired styling to this dialog, which is an anti-pattern.  Ideally, we'd be moving this dialog towards being configurable by its various clients.  If we do want the option to make the entire dialog content scrollable, then I think it should be a configurable option, rather than a default.

**Comment by levadadenys** [comment]
> Some thoughts:
> 
> * I think it might be better for this particular dialog if the buttons at the bottom are always visible, and the body content is scrollable.
> * As a shared component, designed for very widespread use, I'm not sure whether it makes sense to make this always scrollable, given that other users of it might want different behavior.
> * We are already having to use `! important` to [apply](https://github.com/code-dot-org/code-dot-org/blob/902c0f44fad6c1e35f2049dca8f7ad0a9f67a596/apps/src/dance/ai/dance-ai-modal.module.scss#L82-L87) desired styling to this dialog, which is an anti-pattern.  Ideally, we'd be moving this dialog towards being configurable by its various clients.  If we do want the option to make the entire dialog content scrollable, then I think it should be a configurable option, rather than a default.

@breville 
* +1 re making buttons always visible and only content scrollable
* re `!importnat` usage - oh, that actually can be simplified to 
`div.dialog {
  max-width: initial;
  width: 824px;
  height: 494px;
  padding: 0;
  ...`
   instead of `.dialog {`
   Just checked it locally and it works. `AccessibleDialog` wasn't developed in scope of `Design system` so there's no documentation like we have for `componentsLibrary`, but we've faced this issue in typography and this workaround helped there!

**Comment by kobryan0619** [comment]
Just to close the loop here, I talked to Molly about this a bit who shared that the `AccessibleDialog` component will ultimately need a re-do in the future that goes beyond this "scrollable" change.  Based on feedback from @breville @levadadenys we decided to change the scrollable feature in the `MultipleSectionAssigner` vs the `AccessibleDialog` overall.  You can see that in [this PR](https://github.com/code-dot-org/code-dot-org/pull/54808).  Thanks again everyone!

**Pull request closed without merging**